contracts/rosca.sol

```diff
@@ -1,72 +1,266 @@
 pragma solidity ^0.4.4;
 
+/*
+*  Possible missing functions
+* - integrity check? to perform check and make sure certain values are available and correct
+*
+*/
 contract rosca {
-    address WETRUST = 0;  // constant
-    
-    event LogParticipantRegistered(string uniqName);
-    event LogContributionMade(string uniqName);
-    event LogBidPlaced(string uniqName);
-    event LogRoundFundsRelease(string winnerUniqName, uint amountInWei);
-    
-    
-    enum Period { Weekly, Monthly }
-    
+    address constant WETRUST = 0x0;
+
+    enum Period { Weekly, Monthly } // are we only allwing 2 options?
+
+    event LogParticipantRegistered(address user);
+    event LogContributionMade(address user, uint amount);
+    event LogNewLowestBid(uint bid,address winnerAddress);
+    event LogRoundFundsRelease(address winnerUniqName, uint amountInWei);
+
+
+    //added
+    event LogFundsWithdrawal(address user, uint amount,address destination);
+    event LogStartOfPeriod(uint currentRound);
+
+    //state variables
+    address foreman;
+    uint roundSum;
+    uint numRounds;
+    uint currentEpoch;
+    uint numEpochs;
+    int8 minParticipants;
+    uint roundPeriodInDays;
+    uint startTime;
+    int16 feeToService;
+    address feeAddress;
+    uint currentRound;  //currentRound will be set to 0 when ROSCA is created and will turn to one when the ROSCA actually start
+
+    /*
+        not sure about which structures to use yet
+    */
+    struct USER{
+        uint contributed; // number of times contributed or could be amount contributed
+        uint paid; // number of times bid Won
+        uint allowance; // how much they are allowed to withdraw, i.e if someone win , their allowance will go up by the bid.
```

Comment:
nit: in some other contracts, I saw this named pendingWithdrawl . We might want to use the same name (as allowance has some other meanings too)